resource/js/vocab-search.js

```diff
@@ -7,18 +7,119 @@ const vocabSearch = Vue.createApp({
       languages: [],
       selectedLanguage: null,
       searchTerm: null,
-      autoCompeteResults: [],
-      languageStrings: null
+      searchCounter: null,
+      renderedResultsList: [],
+      languageStrings: null,
+      msgs: null
     }
   },
   mounted () {
     this.languages = SKOSMOS.languageOrder
     this.selectedLanguage = SKOSMOS.content_lang
-    this.languageStrings = SKOSMOS.language_strings[SKOSMOS.lang]
+    this.searchCounter = 0
+    this.languageStrings = SKOSMOS.language_strings[SKOSMOS.lang] ?? SKOSMOS.language_strings.en
+    this.msgs = SKOSMOS.msgs[SKOSMOS.lang] ?? SKOSMOS.msgs.en
+    this.renderedResultsList = []
+    document.addEventListener('click', this.onClickOutside)
+  },
+  beforeUnmount () {
+    document.removeEventListener('click', this.onClickOutside)
   },
   methods: {
     autoComplete () {
+      const delayMs = 300
+
+      // when new autocomplete is fired, empty the previous result
+      this.hideDropdown()
+
+      // cancel pending API calls when method is called
+      clearTimeout(this._timerId)
+
+      // is the search term is in cache, use the cache
+
+      // delay call, but don't execute if the search term is not at least two characters
+      if (this.searchTerm.length > 1) {
+        this._timerId = setTimeout(() => { this.search() }, delayMs)
+      }
+    },
+    /*
+     * search should fetch the response and save it to cache
+     * search calls renderResults for displaying the response
+     */
+    search () {
+      const mySearchCounter = this.searchCounter + 1 // make sure we can identify this search later in case of several ongoing searches
```

Comment:
This counter logic is one way of avoiding the problems with AJAX race conditions (#1561) seen in partial page loads. Maybe a more general mechanism could handle both these API calls and the partial page loads, as well as other types of dynamic HTTP calls?

Reply:
Yeah, I was thinking of an abort controller -based general solution.